src/app/Dashboard/AddCard.tsx

```diff
@@ -157,3 +233,124 @@ export const AddCard: React.FunctionComponent<AddCardProps> = (props: AddCardPro
     </>
   );
 };
+
+interface PropsConfigFormProps {
+  cardTitle: string;
+  controls: PropControl[];
+  initialState: any;
+  onChange: ({}) => void;
+}
+
+const PropsConfigForm = (props: PropsConfigFormProps) => {
+  const [propsConfig, setPropsConfig] = React.useState(props.initialState);
+
+  const handleChange = React.useCallback(
+    (k) => (e) => {
+      setPropsConfig((old) => {
+        const copy = { ...old };
+        copy[k] = e;
+        props.onChange(copy);
+        return copy;
+      });
+    },
+    [props.onChange, setPropsConfig]
+  );
+
+  const handleNumeric = React.useCallback(
+    (k) => (e) => {
+      const value = (e.target as HTMLInputElement).value;
+      setPropsConfig((old) => {
+        const copy = { ...old };
+        copy[k] = value;
+        props.onChange(copy);
+        return copy;
+      });
+    },
+    [props.onChange, setPropsConfig]
+  );
+
+  const handleNumericStep = React.useCallback(
+    (k, v) => (e) => {
+      setPropsConfig((old) => {
+        const copy = { ...old };
+        copy[k] = old[k] + v;
+        props.onChange(copy);
+        return copy;
+      });
+    },
+    [props.onChange, setPropsConfig]
+  );
+
+  const createControl = React.useCallback(
+    (ctrl: PropControl): JSX.Element => {
+      let input: JSX.Element;
+      switch (ctrl.kind) {
+        case 'boolean':
+          input = (
+            <Switch label={ctrl.name} isReversed isChecked={propsConfig[ctrl.key]} onChange={handleChange(ctrl.key)} />
+          );
+          break;
+        case 'number':
+          input = (
+            <NumberInput
+              inputName={ctrl.name}
+              inputAriaLabel={`${ctrl.name} input`}
+              value={propsConfig[ctrl.key]}
+              onChange={handleNumeric(ctrl.key)}
+              onPlus={handleNumericStep(ctrl.key, 1)}
+              onMinus={handleNumericStep(ctrl.key, -1)}
+            />
+          );
+          break;
+        case 'string':
+          input = (
+            <TextInput
+              type="text"
+              aria-label={`${ctrl.key} input`}
+              value={propsConfig[ctrl.key]}
+              onChange={handleChange(ctrl.key)}
+            />
+          );
+          break;
+        case 'text':
+          input = (
+            <TextArea
+              type="text"
+              aria-label={`${ctrl.key} input`}
+              value={propsConfig[ctrl.key]}
+              onChange={handleChange(ctrl.key)}
+            />
+          );
+          break;
+        default:
+          input = <Text>Bad config</Text>;
+          break;
+      }
+      return (
+        <FormGroup
+          key={`${ctrl.key}}`}
+          label={ctrl.kind !== 'boolean' ? ctrl.name : undefined}
```

Comment:
On this note https://github.com/cryostatio/cryostat-web/pull/747/files#r1046563123, do you think we might need the form label for switch?